hack/generate-gpu-count-table.sh

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+BIN="$(dirname "$(realpath "${BASH_SOURCE[0]}")")/../bin"
+
+function get_gpus_for_region() {
+  REGION="${1}"
+  echo "Getting gpu counts for ${REGION}..." >&2
+  "${BIN}/aws" ec2 describe-instance-types --region "${REGION}" --query "InstanceTypes[?GpuInfo!=null].[InstanceType, GpuInfo]" |
+    jq -r 'map("\"" + .[0] + "\": " + (.[1].Gpus | map(.Count) | add | tostring) + ",") | .[]'
+}
+
+function get_all_gpus() {
+  "${BIN}/aws" account list-regions --max-results 50 | jq -r '.Regions | map(.RegionName) | .[]' | while read REGION; do
+    get_gpus_for_region $REGION
```

Comment:
Even if we can't combine gpu and InstanceStore queries into one, can we parallelize this inner loop? 

**This script goes from 4.25 minutes to 39 seconds with this change.** 

```
"${BIN}/aws" account list-regions --max-results 50 | jq -r '.Regions | map(.RegionName) | .[]' | while read REGION; do
     sleep 1
     get_gpus_for_region $REGION &
    done
```


The `&` "terminates an asynchronous command" such that it will run in background and run next command (next loop iteration) immediately.

The `sleep 1` sleeps for one second, which prevents the rate-limit issues @ConnorJC3 mentioned during demo. (Or atleast I'm not running into any limits locally) 

Same thing for the instance store generation function


Speedup proof:

```
./hack/generate-gpu-count-table.sh  30.72s user 4.08s system 88% cpu 39.507 total
./hack/generate-gpu-count-table.sh  36.98s user 5.20s system 16% cpu 4:16.95 total
```

I also diffed the results and found no difference.